core/src/rpc.rs

```diff
@@ -243,15 +243,19 @@ impl JsonRpcRequestProcessor {
     ) -> RpcResponse<Option<UiAccount>> {
         let config = config.unwrap_or_default();
         let bank = self.bank(config.commitment);
-        let encoding = config.encoding.unwrap_or(UiAccountEncoding::Binary);
+        let encoding = config.encoding.unwrap_or(UiAccountEncoding::Binary64);
         new_response(
             &bank,
             bank.get_account(pubkey).and_then(|account| {
                 if account.owner == spl_token_id_v1_0() && encoding == UiAccountEncoding::JsonParsed
                 {
                     get_parsed_token_account(bank.clone(), account)
                 } else {
-                    Some(UiAccount::encode(account, encoding, None))
+                    if encoding == UiAccountEncoding::Binary && account.data.len() > 128 {
+                        None
```

Comment:
I think we should return an error back over RPC here, so the user knows they got rejected

Reply:
added